numpy/lib/financial.py

```diff
@@ -206,12 +211,12 @@ def pmt(rate, nper, pv, fv=0, when='end'):
     example illustrates usage of `fv` having a default value of 0.
 
     """
-    when = _convert_when(when)
+    when = _convert_when(when, sentinel=rate)
     (rate, nper, pv, fv, when) = map(np.array, [rate, nper, pv, fv, when])
     temp = (1 + rate)**nper
-    mask = (rate == 0.0)
-    masked_rate = np.where(mask, 1.0, rate)
-    z = np.zeros(np.broadcast(masked_rate, nper, pv, fv, when).shape)
+    mask = (rate == 0)
+    masked_rate = np.where(mask, 1, rate)
+    z = np.zeros(np.broadcast(masked_rate, nper, pv, fv, when).shape, dtype=rate.dtype)
```

Comment:
Should probably be `np.result_type(rate, pv, fv)`

Reply:
Should it match the args given to the `np.broadcast` call? 

So it'd be: 

```python
z = np.zeros(np.broadcast(masked_rate, nper, pv, fv, when).shape,
             dtype=np.result_type(masked_rate, nper, pv, fv, when))
```